pontoon/tags/utils/stats.py

```diff
@@ -0,0 +1,57 @@
+
+from django.db.models import F, Sum, Value
+from django.db.models.functions import Coalesce
+
+from .base import TagsTRTool
+
+
+class TagsStatsTool(TagsTRTool):
+    """Creates aggregated stat data for tags according to
+    filters
+    """
+
+    coalesce = list
+
+    filter_methods = (
+        'tag', 'projects', 'locales', 'path')
+
+    # from the perspective of translated resources
+    _default_annotations = (
+        ('total_strings', Coalesce(
+            Sum('resource__total_strings'),
+            Value(0))),
+        ('fuzzy_strings', Coalesce(
+            Sum("fuzzy_strings"),
+            Value(0))),
+        ('approved_strings', Coalesce(
+            Sum("approved_strings"),
+            Value(0))),
+        ('translated_strings', Coalesce(
+            Sum("translated_strings"),
+            Value(0))))
+
+    def get_data(self):
```

Comment:
Can we get some comments in here explaining what's going on?

Reply:
yep